File: IT/Strutture dati/Esercizi/Java/Esercizio18.java
/* file Esercizio18.java */

import java.util.Arrays;

public class Esercizio18 {
	public static void main(String [] args) {
		int [] a = {1, 2, 3};
		int [] b = {3, 2, 1};
		int [] c = {1, 2, 3};

		System.out.println("a = " + Arrays.toString(a));
		System.out.println("b = " + Arrays.toString(b));
		System.out.println("c = " + Arrays.toString(c));

		System.out.println("a e b sono uguali? " + Arrays.equals(a, b));
		System.out.println("a e c sono uguali? " + Arrays.equals(a, c));
	}
}

File: IT/Strutture dati/Esercizi/C++/esercizio18.cpp
/*
file esercizio18.cpp
compilare con: g++ -std=c++11 esercizio18.cpp
*/

#include <iostream>
#include <iterator>
#include <array>

using std::cout;
using std::endl;
using std::array;
using std::ostream_iterator;
using std::boolalpha;

int main()
{
	array<int, 3> a = {1, 2, 3};
	array<int, 3> b = {3, 2, 1};
	array<int, 3> c = {1, 2, 3};

	cout << "a = [";
	copy(a.begin(), a.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "b = [";
	copy(b.begin(), b.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "c = [";
	copy(c.begin(), c.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "a e b sono uguali? " << boolalpha << (a == b) << endl;
	cout << "a e c sono uguali? " << boolalpha << (a == c) << endl;
}

